extension/headerssetterextension/config.go

```diff
@@ -33,25 +34,51 @@ type Config struct {
 }
 
 type HeaderConfig struct {
-	Key         *string `mapstructure:"key"`
-	Value       *string `mapstructure:"value"`
-	FromContext *string `mapstructure:"from_context"`
+	Action      actionValue `mapstructure:"action"`
+	Key         *string     `mapstructure:"key"`
+	Value       *string     `mapstructure:"value"`
+	FromContext *string     `mapstructure:"from_context"`
 }
 
+// actionValue is the enum to capture the four types of actions to perform on a header
+type actionValue string
+
+const (
+	// INSERT inserts the new header if it does not exist
+	INSERT actionValue = "insert"
+
+	// UPDATE updates the header value if it exists
+	UPDATE actionValue = "update"
+
+	// UPSERT inserts a header if it does not exist and updates the header
+	// if it exists
+	UPSERT actionValue = "upsert"
+
+	// DELETE deletes the header
+	DELETE actionValue = "delete"
+)
+
 // Validate checks if the extension configuration is valid
 func (cfg *Config) Validate() error {
 	if cfg.HeadersConfig == nil || len(cfg.HeadersConfig) == 0 {
 		return errMissingHeadersConfig
 	}
 	for _, header := range cfg.HeadersConfig {
+		if header.Action == "" {
```

Comment:
This makes the extension backward incompatible. You can assume that all existing actions are "insert", and you'll be backward compatible.

Reply:
It is still `alpha`, so there was no intention to stay compatible. Therefore, I left a note in the change log about changing the existing config. Also, I wanted to require the action explicitly without taking `insert` or `upsert` as a default action.

Reply:
maybe then `change_type: 'enhancement'` should be `breaking`?

Reply:
Can we do this in two steps then? The first is non-breaking, assuming that 'insert' is the default action, and will log a WARN if no action is defined, then on N+2, we fail if no action is provided. 

This way, current users won't be caught by surprise when they just bump the collector version: we warned them via changelog *and* via a WARN log entry on their collectors for at least one version.